src/Symfony/Component/DependencyInjection/Loader/Configurator/AbstractConfigurator.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Loader\Configurator;
+
+use Symfony\Component\DependencyInjection\Definition;
+
+abstract class AbstractConfigurator
+{
+    const FACTORY = 'unknown';
+
+    protected $parent;
+    protected $definition;
+    protected $id;
+    protected $defaultTags = array();
+
+    public function __construct(AbstractConfigurator $parent, Definition $definition, $id = null, array $defaultTags = array())
+    {
+        $this->parent = $parent;
+        $this->definition = $definition;
+        $this->id = $id;
+        $this->defaultTags = $defaultTags;
+    }
+
+    public function __destruct()
+    {
+        // default tags should be added last
+        foreach ($this->defaultTags as $name => $attributes) {
+            foreach ($attributes as $attributes) {
+                $this->definition->addTag($name, $attributes);
+            }
+        }
+        $this->defaultTags = array();
+    }
```

Comment:
Actually, I can't find where this is used. `$this->defaultTags` is only set in `__construct()`, but the constructor is overridden in almost all classes that extend this. I could be wrong... but is this used anymore?

Reply:
destruct is the correct operation - php doesn't do it but it could here - and it's required for us.
adding a method would mean polluting autocompletion also